vue/src/apps/RecipeEditView/RecipeEditView.vue

```diff
@@ -1065,6 +1080,8 @@ export default {
                 always_use_plural_food: false,
                 original_text: null,
             })
+            // TODO: probably can remove this if find more elegant way to have the initial ingredient's show_ingredients_table set to the user_preference value
+            step.show_ingredients_table = this.user_preferences.show_step_ingredients
```

Comment:
I couldn't figure out how to modify the state of the initial step of a manually created recipe. So I've patched `addIngredient`, so that the initial step will reflect a user's default preferences. However, this isn't great because if you edit a recipe step where you're happy with the setting not being the default, adding a new ingredient will switch the setting to the default setting. 

@vabene1111 any pointers on where I can set state for the first step in a manually created recipe?

Reply:
Hi, the first step of a new recipe is created here https://github.com/TandoorRecipes/recipes/blob/develop/cookbook/views/new.py#L40

its an old relict but it still works, so there you can just hook into the user setting and create the step accordingly